ddtrace/tracer/span.go

```diff
@@ -441,11 +446,15 @@ func (s *Span) setTagError(value interface{}, cfg errorConfig) {
 			s.setMeta(ext.ErrorDetails, fmt.Sprintf("%+v", v))
 			if !cfg.noDebugStack {
 				s.setMeta(ext.ErrorStack, err.Format())
+				s.setMeta(ext.ErrorHandlingStack, takeStacktrace(cfg.stackFrames, cfg.stackSkip))
 			}
 			return
 		}
 		if !cfg.noDebugStack {
-			s.setMeta(ext.ErrorStack, takeStacktrace(cfg.stackFrames, cfg.stackSkip))
+			telemetry.Count(telemetry.NamespaceTracers, "errorstack.source", []string{"source:takeStacktrace"}).Submit(1)
+			stack := takeStacktrace(cfg.stackFrames, cfg.stackSkip)
+			s.setMeta(ext.ErrorStack, stack)
```

Comment:
Maybe we shouldn't set the error stack here to indicate we actually haven't capture the "real" stack trace of the error.

Reply:
I actually think it's the other way around (`!cfg.noDebugStack` means we're not disabling the debug stack, so we _should_ capture it)?

Reply:
Oh, yeah. It should be an xor between those two.